database/buildSchema/18_application.sql

```diff
@@ -55,3 +55,49 @@ CREATE TRIGGER outcome_trigger
     WHEN (OLD.outcome = 'PENDING' AND NEW.outcome <> 'PENDING')
     EXECUTE FUNCTION public.outcome_changed ();
 
+--FUNCTION to revert application status/active when OUTCOME is changed back to PENDING
+CREATE OR REPLACE FUNCTION public.outcome_reverted ()
+    RETURNS TRIGGER
+    AS $application_event$
+BEGIN
+    UPDATE
+        public.application
+    SET
+        is_active = TRUE
+    WHERE
+        id = NEW.id;
+    INSERT INTO public.application_status_history (application_stage_history_id, status)
+        VALUES ((
+                SELECT
+                    id
+                FROM
+                    application_stage_history
+                WHERE
+                    application_id = NEW.id
+                    AND is_current = TRUE),
+                (
+                    SELECT
+                        status
+                    FROM
+                        application_status_history
+                    WHERE
+                        time_created = (
```

Comment:
Assuming that the "previous" record will be the one with the most recent timestamp that is currently `is_current = false`